File: 240302/그 요일은/the-day-of-the-day.py
dates = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
m1, d1, m2, d2 = tuple(map(int, input().split()))
day = input()

def num_of_dates(m,d):
    total = 0

    for i range(1, m):
        total += dates[i]

    return total += d

def num_of_day(s):
    return days[s]

ans = 0
start_date = num_of_dates(m1, d1)
end_date = num_of_dates(m2, d2)
cur_day = num_of_day("Mon")

for date in range(start_date, end_date + 1):
    if(cur_day == num_of_day(day)):
        ans += 1
    
    cur_day = (cur_day + 1) % 7

print(cur_day)

File: 240302/그 요일은/the-day-of-the-day.js
//                 1.  2.  3.  4.  5.  6.  7.  8.  9.  10. 11. 12.
const MONTHS = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
const DAYS = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]

const fs = require("fs");
let [dates, _day] = fs.readFileSync(0).toString().trim().split("\n");
const day = _day.trim()
const [m1, d1, m2, d2] = dates.trim().split(" ").map(Number);
let totalDays = 1;

for (let month = m1; month < m2; month++) {
    totalDays += MONTHS[month];
}
totalDays -= d1;
totalDays += d2;

function f(n) {
    if(DAYS[n] === day) return true
    else if(n % 7 === n) return false
    else return f(n % 7)
}

let cnt = 0;
for(let i = totalDays; 0 < i; i--) {
    if(f(i - 1)) {
        cnt++;
    }
}

console.log(cnt)